files/zh-cn/learn/javascript/building_blocks/functions/index.md

```diff
@@ -127,132 +120,195 @@ myFunction();
+### 箭头函数
+
+如果你传递这样一个匿名函数，你可以使用另一种形式，即箭头函数。你可以写 `(event) =>` 来代替 `function(event)`：
 
 ```js
-myButton.onclick = function () {
-  alert("hello");
-  // 我可以在这里放入
-  // 尽可能多的代码
-};
+textBox.addEventListener("keydown", (event) => {
+  console.log(`You pressed "${event.key}".`);
+});
 ```
 
-> **备注：** 匿名函数也称为函数表达式。函数表达式与函数声明有一些区别。函数声明会进行声明提升（declaration hoisting），而函数表达式不会。
+如果函数只接受一个参数，可以省略参数周围的括号：
 
-## 函数参数
+```js
+textBox.addEventListener("keydown", event => {
+  console.log(`You pressed "${event.key}".`);
+});
+```
 
-一些函数需要在调用它们时指定参数——这些参数值需要放在函数括号内，才能正确地完成其工作。
+最后，如果函数只包含一行 `return` 语句，也可以省略圆括号和 `return` 关键字，隐式地返回表达式。在下面的示例中，我们使用 `Array` 的 {{jsxref("Array.prototype.map()","map()")}} 方法将原始数组中的每个值加倍：
 
-> **备注：** 参数有时称为参数（argument）、属性（propertie）或甚至特性（attribute）。
+```js
+const originals = [1, 2, 3];
 
-例如，浏览器的内置 [Math.random()](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Math/random) 函数不需要任何参数。当被调用时，它总是返回 0 到 1 之间的随机数：
+const doubled = originals.map(item => item * 2);
+
+console.log(doubled); // [2, 4, 6]
+```
+
+`map()` 方法依次获取数组中的每一项，并将其传递给给定函数。然后，它将该函数返回的值添加到一个新数组中。
+
+因此，在上面的例子中，`item => item * 2` 相当于箭头函数：
 
 ```js
-var myNumber = Math.random();
+function doubleItem(item) {
+  return item * 2;
+}
 ```
 
-浏览器的内置字符串[replace（）](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String/replace)函数需要两个参数：在主字符串中查找的子字符串，以及用以下替换该字符串的子字符串：
+你可以使用同样简洁的语法重写 `addEventListener` 示例。
 
 ```js
-var myText = "I am a string";
-var newString = myText.replace("string", "sausage");
+textBox.addEventListener("keydown", (event) =>
+  console.log(`You pressed "${event.key}".`),
+);
 ```
 
-> **备注：** 当你需要指定多个参数时，它们以逗号分隔。
+我们建议你使用箭头函数，因为它们可以使你的代码更简短、更易读。要了解更多信息，请参阅 [JavaScript 指南中的有关部分](/zh-CN/docs/Web/JavaScript/Guide/Functions#arrow_functions)，以及我们[有关箭头函数的参考文档](/zh-CN/docs/Web/JavaScript/Reference/Functions/Arrow_functions)。
+
+> **备注：** 注意：箭头函数和普通函数之间存在一些细微差别。它们超出了本入门指南的范围，而且在我们这里讨论的情况下不太可能有什么不同。要了解更多信息，请参阅 [箭头函数参考文档](/zh-CN/docs/Web/JavaScript/Reference/Functions/Arrow_functions)。
+
+### 箭头函数举例
 
-还应该注意，有时参数不是必须的——你不必指定它们。如果没有，该功能一般会采用某种默认行为。作为示例，数组 [`join()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/join) 函数的参数是可选的：
+下面是我们上面讨论的“keydown”示例的完整工作示例：
 
-<!-- autocorrect-disable -->
+HTML 文件：
+
+```html
+<input id="textBox" type="text" />
+<div id="output"></div>
+```
+
+JavaScript 文件：
 
 ```js
-const myArray = ["我", "爱", "巧克力", "青蛙"];
-const madeAString = myArray.join(" ");
-console.log(madeAString);
-// 返回 '我 爱 巧克力 青蛙'
+const textBox = document.querySelector("#textBox");
+const output = document.querySelector("#output");
 
-const madeAnotherString = myArray.join();
-console.log(madeAnotherString);
-// 返回 '我,爱,巧克力,青蛙'
+textBox.addEventListener("keydown", (event) => {
+  output.textContent = `You pressed "${event.key}".`;
+});
+```
+
+```css hidden
+div {
+  margin: 0.5rem 0;
+}
 ```
 
-<!-- autocorrect-enable -->
+结果——试着输入文本框，看看输出结果：
 
-如果没有包含参数来指定加入/分隔符，默认情况下会使用逗号。
+{{EmbedLiveSample("Arrow function live sample", 100, 100)}}
```

Comment:
Match the heading (see [translation-guide](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E7%BF%BB%E8%AF%91%E7%A4%BA%E4%BE%8B%E4%BB%A3%E7%A0%81%E5%9D%97)):

```suggestion
{{EmbedLiveSample("箭头函数实时示例", 100, 100)}}
```